alacritty/src/config/bindings.rs

```diff
@@ -61,16 +61,21 @@ impl<T: Eq> Binding<T> {
             return false;
         }
 
-        // Completely empty modes match all modes.
-        if (self.mode.is_empty() && self.notmode.is_empty())
-            || (binding.mode.is_empty() && binding.notmode.is_empty())
-        {
-            return true;
+        let selfmode = if self.mode.is_empty() { TermMode::ANY } else { self.mode };
+
+        let bindingmode = if binding.mode.is_empty() { TermMode::ANY } else { binding.mode };
+
+        if !selfmode.intersects(bindingmode) {
```

Comment:
```suggestion
        // If none of the required modes intersect, we can be confident that the bindings are not in conflict, since they will never be active at the same time.
        if !selfmode.intersects(bindingmode) {
```